packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+const double _kDefaultSubmenuIconSize = 24.0;
+
+// The default spacing between the the leading icon, label, trailing icon, and
+// shortcut label in a _MenuItemLabel.
+const double _kLabelItemDefaultSpacing = 18.0;
+
+// The minimum spacing between the the leading icon, label, trailing icon, and
+// shortcut label in a _MenuItemLabel.
+const double _kLabelItemMinSpacing = 4.0;
+
+// The minimum horizontal spacing on the outside of the top level menu.
+const double _kTopLevelMenuHorizontalMinPadding = 4.0;
+
+// Navigation shortcuts that we need to make sure are active when menus are
+// open.
+const Map<ShortcutActivator, Intent> _kMenuTraversalShortcuts = <ShortcutActivator, Intent>{
+  SingleActivator(LogicalKeyboardKey.gameButtonA): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.escape): DismissIntent(),
+  SingleActivator(LogicalKeyboardKey.tab): NextFocusIntent(),
+  SingleActivator(LogicalKeyboardKey.tab, shift: true): PreviousFocusIntent(),
+  SingleActivator(LogicalKeyboardKey.arrowDown): DirectionalFocusIntent(TraversalDirection.down),
+  SingleActivator(LogicalKeyboardKey.arrowUp): DirectionalFocusIntent(TraversalDirection.up),
+  SingleActivator(LogicalKeyboardKey.arrowLeft): DirectionalFocusIntent(TraversalDirection.left),
+  SingleActivator(LogicalKeyboardKey.arrowRight): DirectionalFocusIntent(TraversalDirection.right),
+};
+
+/// A menu bar that manages cascading child menus.
+///
+/// This is a Material Design menu bar that typically resides above the main
+/// body of an application (but can go anywhere) that defines a menu system for
+/// invoking callbacks or firing [Intent]s in response to user selection of a
+/// menu item.
+///
+/// The menus can be opened with a click or tap. Once a menu is opened, it can
+/// be navigated by using the arrow and tab keys or via mouse hover. Selecting a
+/// menu item can be done by pressing enter, or by clicking or tapping on the
+/// menu item. Clicking or tapping on any part of the user interface that isn't
+/// part of the menu system controlled by the same controller will cause all of
+/// the menus controlled by that controller to close, as will pressing the
+/// escape key.
+///
+/// {@template flutter.material.menu_bar.shortcuts_note}
+/// Menu items using [MenuItemButton] can have a [SingleActivator] or
+/// [CharacterActivator] assigned to them as their [MenuItemButton.shortcut],
+/// which will display an appropriate shortcut hint. Shortcuts are not
+/// automatically registered, they must be available in the context that the
+/// [MenuBar] resides in, and registered via another mechanism.
+///
+/// If shortcuts should be generally enabled, but are not easily defined in the
+/// context surrounding the menu bar, consider using a [ShortcutRegistry] to
+/// register them. To be sure that selecting a menu item and triggering the
+/// shortcut do the same thing, it is recommended that they trigger the same
+/// [Intent] or call the same callback.
+/// {@endtemplate}
```

Comment:
Should the link to the example also appear here, so that it's picked up wherever this template is used?